website/content/en/preview/upgrading/v1-migration.md

```diff
@@ -0,0 +1,247 @@
+    export CLUSTER_NAME="${USER}-karpenter-demo"
+    export AWS_REGION="us-west-2"
+    export AWS_ACCOUNT_ID="$(aws sts get-caller-identity --query Account --output text)"
+    export KARPENTER_IAM_ROLE_ARN="arn:${AWS_PARTITION}:iam::${AWS_ACCOUNT_ID}:role/${CLUSTER_NAME}-karpenter"
+    export CLUSTER_ENDPOINT="$(aws eks describe-cluster --name ${CLUSTER_NAME} --query "cluster.endpoint" --output text)"
+    ```
+
+4. Apply the new Karpenter policy and assign it to the existing Karpenter role:
+
+    ```bash
+    TEMPOUT=$(mktemp)
+    curl -fsSL https://raw.githubusercontent.com/aws/karpenter-provider-aws/v0.32.10/website/content/en/preview/upgrading/v1beta1-controller-policy.json > ${TEMPOUT}
+
+    AWS_REGION=${AWS_REGION:=$AWS_DEFAULT_REGION} # use the default region if AWS_REGION isn't defined
+    POLICY_DOCUMENT=$(envsubst < ${TEMPOUT})
+    POLICY_NAME="KarpenterControllerPolicy-${CLUSTER_NAME}-v1beta1"
+    ROLE_NAME="${CLUSTER_NAME}-karpenter"
+
+    POLICY_ARN=$(aws iam create-policy --policy-name "${POLICY_NAME}" --policy-document "${POLICY_DOCUMENT}" | jq -r .Policy.Arn)
+    aws iam attach-role-policy --role-name "${ROLE_NAME}" --policy-arn "${POLICY_ARN}"
+    ```
+
+5. Apply the v1.0.0 Custom Resource Definitions (CRDs):
+
+   ?????  I'm not able to try this since CRDs don't seem to exist yet.  ?????
+
+   ```bash
+   kubectl apply -f https://raw.githubusercontent.com/aws/karpenter-provider-aws/v1.0.0/pkg/apis/crds/karpenter.sh_nodepools.yaml
+   kubectl apply -f https://raw.githubusercontent.com/aws/karpenter-provider-aws/v1.0.0/pkg/apis/crds/karpenter.sh_nodeclaims.yaml
+   kubectl apply -f https://raw.githubusercontent.com/aws/karpenter-provider-aws/v1.0.0/pkg/apis/crds/karpenter.k8s.aws_ec2nodeclasses.yaml
+    ```
+
+6. Upgrade Karpenter to the new version:
+   ????? Assuming the karpenter version is 1.0.0, it is not yet available to test.  ?????
+
+    ```bash
+    helm upgrade --install karpenter oci://public.ecr.aws/karpenter/karpenter --version ${KARPENTER_VERSION} --namespace kube-system --create-namespace \
+      --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"=${KARPENTER_IAM_ROLE_ARN} \
+      --set settings.aws.defaultInstanceProfile=KarpenterNodeInstanceProfile-${CLUSTER_NAME} \
+      --set settings.clusterName=${CLUSTER_NAME} \
+      --set settings.interruptionQueue=${CLUSTER_NAME} \
+      --set controller.resources.requests.cpu=1 \
+      --set controller.resources.requests.memory=1Gi \
+      --set controller.resources.limits.cpu=1 \
+      --set controller.resources.limits.memory=1Gi \
+      --wait
+    ```
+
+   {{% alert title="Note" color="warning" %}}
+   Karpenter has deprecated and moved a number of Helm values as part of the v1 release. Ensure that you upgrade to the newer version of these helm values during your migration to v1. You can find detail for all the settings that were moved in the [v1 Upgrade Reference]({{<ref "#helm-values" >}}).
+   {{% /alert %}}
+
```

Comment:
Can we metntion that there will be a post install hook that will run and configure the conversion webhooks on the custom resources definition as part of the helm process. https://github.com/aws/karpenter-provider-aws/blob/main/charts/karpenter/templates/post-install-hook.yaml. Can we also include customer can run with `--no-hooks` flag to prevent the post install hook from running in their cluster.

Reply:
Done.